### ISSUE TITLE
[V2][DX9] Shader errors in 'Hidden/PostProcessing/DepthOfField' on build

### ISSUE THREAD

**Issue by j4james** [Unity-Technologies/PostProcessing#313]
##### What happened?

When I try to build the player for the Windows target platform I get a number of errors reported in the DepthOfField shader. There are also a few shader warnings showing up too.

```
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(153) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(161) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(162) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(164) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(165) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(169) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(175) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(179) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(182) (on d3d9)
Shader error in 'Hidden/PostProcessing/DepthOfField': maximum temp register index exceeded at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(183) (on d3d9)
Shader warning in 'Hidden/PostProcessing/DepthOfField': floating point division by zero at Assets/PostProcessing/Shaders/Builtins/DepthOfField.hlsl(162) (on d3d9)
Shader warning in 'Hidden/PostProcessing/CopyStd': value cannot be NaN, isnan() may not be necessary.  /Gis may force isnan() to be performed at line 57 (on d3d9)
Shader warning in 'Hidden/PostProcessing/CopyStd': value cannot be infinity, isinf() may not be necessary.  /Gis may force isinf() to be performed at line 57 (on d3d9)
Shader warning in 'Hidden/PostProcessing/Debug/Overlays': value cannot be NaN, isnan() may not be necessary.  /Gis may force isnan() to be performed at line 172 (on d3d9)
Shader warning in 'Hidden/PostProcessing/Debug/Overlays': value cannot be infinity, isinf() may not be necessary.  /Gis may force isinf() to be performed at line 172 (on d3d9)
Shader warning in 'Hidden/PostProcessing/Debug/Overlays': value cannot be NaN, isnan() may not be necessary.  /Gis may force isnan() to be performed at line 172 (on d3d11)
Shader warning in 'Hidden/PostProcessing/Copy': value cannot be NaN, isnan() may not be necessary.  /Gis may force isnan() to be performed at line 21 (on d3d9)
Shader warning in 'Hidden/PostProcessing/Copy': value cannot be infinity, isinf() may not be necessary.  /Gis may force isinf() to be performed at line 21 (on d3d9)
UnityEditor.BuildPlayerWindow+BuildMethodException: Build failed with errors.
  at UnityEditor.BuildPlayerWindow+DefaultBuildMethods.BuildPlayer (BuildPlayerOptions options) [0x001b9] in C:\buildslave\unity\build\Editor\Mono\BuildPlayerWindowBuildMethods.cs:162 
  at UnityEditor.BuildPlayerWindow.CallBuildMethods (Boolean askForBuildLocation, BuildOptions defaultBuildOptions) [0x00050] in C:\buildslave\unity\build\Editor\Mono\BuildPlayerWindowBuildMethods.cs:83 
UnityEditor.HostView:OnGUI()
```
If I disable _Auto Graphics API for Windows_ in the player settings, and remove the _Direct3D9_ API from the supported list, then the build does complete successfully. Is that intentional? Is D3D9 not supported?

##### Post-processing stack version (v1, v2)?
v2

##### Unity version, operating system, target platform (standalone windows, mac, iOS, PS4...)?
Unity 2017.1.0p4 Personal (64bit), Windows 10, Standalone Windows x86

**Comment by Kink3d** [comment]
Hi,

This isnt intentional, but DX9 has some tricky limitations :)

Were going to look at clamping kernel for DoF but this will lead to visual inconsistency with other APIs. Also it might not work and we may have to just handle the errors better. It should be noted that DX9 is not high priority to support for us as we are deprecating support for it in 2017.3

Thanks for reporting

**Comment by j4james** [comment]
Thanks for the response. And FWIW, I'm OK with the DoF effect just not doing anything on DX9. Dropping DX9 support completely is also not the end of the world, if that's the long term plan anyway.

**Comment by empika** [comment]
Is there any way to temporarily fix/disable this? I don't use this effect and the errors are messing with my build scripts. Thanks!

**Comment by SamVickery91** [comment]
For those having issues just disable DX9 support in player settings > Auto Graphics API for Windows

**Comment by empika** [comment]
Ah, of course. Thanks!

**Comment by Chman** [comment]
Fix is in 695dd30.
